core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -214,51 +214,48 @@ DETRAY_HOST_DEVICE void detray::rk_stepper<
+    /*-----------------------------------------------------------------
+     * Calculate the first terms of dk_n/dt1
+    -------------------------------------------------------------------*/
+    // dk1/dt1
+    dkndt[0u] =
+        sd.qop[0u] * mat_helper().column_wise_cross(dkndt[0u], sd.b_first);
+
+    // dk2/dt1
+    dkndt[1u] = dkndt[1u] + half_h * dkndt[0u];
+    dkndt[1u] =
+        sd.qop[1u] * mat_helper().column_wise_cross(dkndt[1u], sd.b_middle);
+
+    // dk3/dt1
+    dkndt[2u] = dkndt[2u] + half_h * dkndt[1u];
+    dkndt[2u] =
+        sd.qop[2u] * mat_helper().column_wise_cross(dkndt[2u], sd.b_middle);
+
+    // dk4/dt1
+    dkndt[3u] = dkndt[3u] + h * dkndt[2u];
+    dkndt[3u] =
+        sd.qop[3u] * mat_helper().column_wise_cross(dkndt[3u], sd.b_last);
+
+    /*-----------------------------------------------------------------
+     * Calculate the first and second terms of dk_n/dqop1
+    -------------------------------------------------------------------*/
+    // dk1/dqop1
+    dkndqop[0u] = dqopn_dqop[0u] * vector::cross(sd.t[0u], sd.b_first);
+
+    // dk2/dqop1
+    dkndqop[1u] = dqopn_dqop[1u] * vector::cross(sd.t[1u], sd.b_middle) +
+                  sd.qop[1u] * half_h * vector::cross(dkndqop[0u], sd.b_middle);
+
+    // dk3/dqop1
+    dkndqop[2u] = dqopn_dqop[2u] * vector::cross(sd.t[2u], sd.b_middle) +
+                  sd.qop[2u] * half_h * vector::cross(dkndqop[1u], sd.b_middle);
+
+    // dk4/dqop1
+    dkndqop[3u] = dqopn_dqop[3u] * vector::cross(sd.t[3u], sd.b_last) +
+                  sd.qop[3u] * h * vector::cross(dkndqop[2u], sd.b_last);
 
-        // dk3/dt1
-        dkndt[2u] = dkndt[2u] + half_h * dkndt[1u];
-        dkndt[2u] =
-            sd.qop[2u] * mat_helper().column_wise_cross(dkndt[2u], sd.b_middle);
-
-        // dk4/dt1
-        dkndt[3u] = dkndt[3u] + h * dkndt[2u];
-        dkndt[3u] =
-            sd.qop[3u] * mat_helper().column_wise_cross(dkndt[3u], sd.b_last);
-
-        /*-----------------------------------------------------------------
-         * Calculate the first and second terms of dk_n/dqop1
-        -------------------------------------------------------------------*/
-        // dk1/dqop1
-        dkndqop[0u] = dqopn_dqop[0u] * vector::cross(sd.t[0u], sd.b_first);
-
-        // dk2/dqop1
-        dkndqop[1u] =
-            dqopn_dqop[1u] * vector::cross(sd.t[1u], sd.b_middle) +
-            sd.qop[1u] * half_h * vector::cross(dkndqop[0u], sd.b_middle);
-
-        // dk3/dqop1
-        dkndqop[2u] =
-            dqopn_dqop[2u] * vector::cross(sd.t[2u], sd.b_middle) +
-            sd.qop[2u] * half_h * vector::cross(dkndqop[1u], sd.b_middle);
-
-        // dk4/dqop1
-        dkndqop[3u] = dqopn_dqop[3u] * vector::cross(sd.t[3u], sd.b_last) +
-                      sd.qop[3u] * h * vector::cross(dkndqop[2u], sd.b_last);
-    } else {
+    // Calculate in the case of not considering B field gradient
+    if (cfg.use_field_gradient) {
```

Comment:
The comment says, that this section is for "not considering the B field gradient". Can you update it or is the if statement wrong?

Reply:
Right. I fixed the comment accordingly